src/pocketmine/Player.php

```diff
@@ -1378,7 +1385,7 @@ protected function checkNearEntities($tickDiff){
 	}
 
 	protected function processMovement($tickDiff){
-		if(!$this->isAlive() or !$this->spawned or $this->newPosition === null or $this->teleportPosition !== null or $this->isSleeping()){
```

Comment:
`for most intents and purposes`

Spectators share many behavioural characteristics with dead players. Movement is not one of them.